addons/website/static/src/snippets/s_popup/000.js

```diff
@@ -142,6 +143,21 @@ const PopupWidget = publicWidget.Widget.extend({
         }
         this.$el.find('.modal').modal('show');
     },
+    /**
+     * Checks if the given primary button should allow or not to close the
+     * modal.
+     *
+     * @private
+     */
+    _canBtnPrimaryClosePopup(primaryBtnEl) {
+        if (
+            primaryBtnEl.classList.contains('s_website_form_send')
+            || primaryBtnEl.classList.contains('o_website_form_send') // !compatibility
+        ) {
+            return false;
+        }
+        return true;
```

Comment:
About the part of the change to have operator at the end of the line instead of at the beginning, it's a terrible idea and really bad, I won't do that part.

It's concerning that the linter is not happy about it :grimacing: 

It makes the condition impossible to read in many cases: you have to go to the end of line to see what is the operator between the 2 part of the condition, and those lines might not even have the same length, requiring a bit of "machine à écrire" effect like reading commit message not wrap at 72/80 char.

It's kind of how it's enforced in python pep8, see https://pep8.org/#break-before-or-after-binary-operator which explains very well the issue to have operator at end of line.


> The code here works and could have advantages that I didn't notice. (History maybe?)

Just better readability IMHO, it kinda reads as easy as in real life: `if I am X or Y I return false` while with the other it's `I return the opposite of if I am X or Y`, probably just matter of preference anyway, I applied the `!(..)` change.

Reply:
> About the part of the change to have operator at the end of the line instead of at the beginning, it's a terrible idea and really bad, I won't do that part.

Damn, I didn't know a linter rule could be that bad 😆 I agree with the pep8 documentation though.

As I said, it's aesthetics, it doesn't matter much to me. Only reason I bring this up is because I use [this tool](https://github.com/odoo/odoo/wiki/Javascript-coding-guidelines) to write new OWL code as they have a few rules specific to OWL components that our old ESLint config did not have. But it's a legacy widget here so these rules matters even less.

Thanks for applying my suggestion though!

Reply:
> Damn, I didn't know a linter rule could be that bad :laughing:

Well, it's obviously just my (strong) opinion, on top of PEP8 one. Some might disagree, it's probably a matter of preference even if I really can't see how someone would argue against this in favor of the opposite as it really makes everything easier to read.

Reply:
Note that it's always a though topic (I really can't understand why tho):
- Airbnb also enforced the operator at the start of the line as in pep8
  https://github.com/airbnb/javascript/commit/2ab0e618582f5b64e406fe81fb5c9540b3be9824 (coming from https://github.com/airbnb/javascript/issues/1380) 
- It's actually the same in SQL
```sql
SELECT id
  FROM ir_ui_view
 WHERE key = 'website.layout'
   AND (website_id IS NULL
    OR website_id = %(website_id)s)
```

Reply:
Just to be clear, I do not argue about this specific rule. I agree with you that it makes more sense to write with operators at the beginning, especially now that you've brought a few good arguments to the table.

However, I think right now, the situation is quite bad:
- Our team uses the old linter which does not have any OWL rule
- Externals use the /web linter which disabled itself on website modules
- It's unclear what rule to follow for website in what context.

The point I was trying to raise was this:
If we use their linter starting with any new line in master, we will eventually reach a state where we can enable the linter on specific files to start, maybe the entire module at some point.

Unfortunately doing that means following some broken rules which make no sense. (IMO single quote for technical and double quote for text made a lot of sense and I wish we could have kept it, but their linter enforce double quote all the time).

I think everyone having the same linter outweighs the bad for multiple reasons but mostly it would fix the pain points I mentioned earlier.

Reply:
Yeah, I understand your point.
It actually make sense. It'd be easier if we just had a few ground rules, and a bunch of "nice to have" rules.

> For tracking purpose, here is my config:

I have the same (or very similar) config for versions prior to master (though I manually change it every time).

> At the end, I am probably just an grumpy annoying developer and what should probably be done is

I think you raise completely valid points. (I'd actually be really happy if we had our own config at the root of our folder or somewhere in there I suppose, we could have the same style everywhere while avoiding rules like the operator at the end of the line). 
In any case, thanks for having an open debate about that with me, it's nice to see differing viewpoints while not feeling completely ignored  :D I'll try to not enforce /web's linter (too much)!

Reply:
> I'll try to not enforce /web's linter (too much)!

Nah it's okay, you can do it, most people have the same linter anyway, it's just going to be difficult to make me comply to it :innocent: .